src/mqtt/client.rs

```diff
@@ -911,7 +911,53 @@ impl asynch::Connection for EspAsyncMqttConnection {
     }
 }
 
-static ERROR: EspError = EspError::from_infallible::<ESP_FAIL>();
+
+#[derive(Debug)]
+pub enum MqttClientError {
```

Comment:
Could you please add a few words of documentation and the link to the error description on the esp-idf docs: <https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/protocols/mqtt.html#_CPPv420esp_mqtt_error_codes>